docs/DeveloperGuide.md

```diff
@@ -137,29 +191,44 @@ How the parsing works:
-* contains the `AddressBook`. 
-  * `AddressBook` stores all tuition address book data, i.e., all `Person`, calendar `Entry` and `Tag` objects (which are contained in a `UniquePersonList`, `CalendarEntryList` and `UniqueTagList` objects respectively).
-  * While `UniqueTagList` stores all existing distinct tags created, `Tag` objects with the same tag name may not have the same reference. This means that each `Person` object still has its own `Tag` objects and the `Tag` is only added into `UniqueTagList` if it does not exist in the tag list yet.
-* stores the currently 'selected' `Person` objects (e.g., results of a search query) as a separate filtered list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed', e.g., the `Ui` component can be bound to this list so that it automatically updates when the data in the list change.
+* stores the tuition address book data, i.e., all `Person`, calendar `Entry`, and `Tag` objects (which are contained in the `UniquePersonList`, `CalendarEntryList`, and `UniqueTagList` objects respectively).
+* stores the currently 'filtered' `Person` objects (e.g., results of a search query) as a separate filtered list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed', e.g., the `Ui` component can be bound to this list so that it automatically updates when the data in the list change.
 * stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
 * does not depend on any of the other three components (as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components).
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** 
+The _Sequence Diagram_ below illustrates the interactions when a person, p, is removed from the model.
+
+![Interactions Inside the Model Component for the `delete 1` Command](images/ModelDeleteSequenceDiagram.png)
+
+*Figure A.3.2: Sequence Diagram of Deleting a Person from the Model*
+
+Note that deleting a person in the tuition address book requires
+1. removing that person from the `UniquePersonList`
+2. removing all calendar `Entry`s related to that person from the `CalendarEntryList`
+3. removing all `Tag`s related to that person from the `UniqueTagList`
+
+Working together, the `UniquePersonList`, `CalendarEntrylist`, and `UniqueTagList` manage all of TAB's data.
+**All three of these lists must be updated together, whenever a change is made to TAB's data.**
+
+<div markdown="span" class="alert alert-info">
+
+:information_source: **Note:** 
+The `UniqueTagList` stores all existing distinct tags created, `Tag` objects with the same tag name may not have the same reference.
+This means that each `Person` object still has its own `Tag` objects and the `Tag` is only added into `UniqueTagList` if a duplicate `Tag` does not already exist in the tag list.
```

Comment:
```suggestion
This means that each `Person` object still has its own `Tag` objects and the `Tag` is only added from the `Person` into `UniqueTagList` if an equivalent `Tag` (with the same tag name) does not exist in the tag list.
```

Reply:
also, i think reminder should come before tag since reminder is contained inside `CalendarEntryList` and is also related to the lesson management